### PR TITLE
Add a contribution guide

### PR DESCRIPTION
**Pull request by kubadz**
**Description**

Changes proposed in this pull request:
- add a contribution guide

**Related issue(s)**

Related to [#kyma/6118](https://github.com/kyma-project/kyma/issues/6118)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+cd dex
+
+# configure the repository
+
+git remote add upstream https://github.com/dexidp/dex.git
+git fetch upstream
+git branch -u upstream/master
+```
+
+## To create a PR to the fork:
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
## Create a pull request
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Contributing to dex fork
+
+As the git workflow we are using in Kyma differs from the workflow in this repo, this readme contains instructions for some basic operations.
+
+## To prepare a local repository:
+
+```shell script
+# clone fork's repository
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Clone the fork repository
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+cd $GOPATH/src/github.com
+mkdir dexidp
+cd dexidp
+git clone https://github.com/kyma-incubator/dex.git
+cd dex
+
+# configure the repository
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Configure the cloned repository
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+git remote add upstream https://github.com/dexidp/dex.git
+git fetch upstream
+git branch -u upstream/master
+```
+
+## To create a PR to the fork:
+
+> always check out from `kyma-master` branch
+
+> make sure to choose `kyma-incubator/dex` repository and `kyma-master` branch as the base for your PR
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]: Replace these lines with: 

```
>**NOTE:** Always check out from the `kyma-master` branch. Chose the `kyma-incubator/dex` repository and the `kyma-master` branch as the base of your PR.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+```shell script
+# fetch recent kyma-master
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Fetch kyma-master
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+git checkout kyma-master
+git pull origin kyma-master
+
+# create your branch with changes
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Create a branch for your changes
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+git checkout -b <your_branch_name>
+
+# ... commit changes ...
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Commit changes to your branch
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+# push your branch
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Push your branch
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+git push origin <your_branch_name>
+
+# create a PR from the browser
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
# Create a PR through the GitHub UI
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+```
+
+## To update kyma-master branch:
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
## Update the kyma-master branch
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+git pull
+```
+
+Then in the place of `# ... commit changes ...` comment, you should merge the `master` to your branch. You have to resolve the conflicts if there are any.
+
+```shell script
+# merge changes from master to your branch
+
+git merge master
+```
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]: This is a bit too complicated. There are basically two things to do to update the `kyma-master`, you don't really have to mention any other instructions here. Please work on this section.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+### To update the remote master branch:
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]: Make this a separate section instead of a sub-section.

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+> you have to be able to push to `master` branch to complete this operation
+
+If you prepared your local repository following the steps above, `git pull` command on `master` branch will fetch changes from the original repository to your local repository. To push it to remote repository simply type `git push origin master`.
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]: Skip the "if" part, let's assume that the reader follows the workflow described in this doc. Focus just on the things to do.

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+```shell script
+# merge changes from master to your branch
+
+git merge master
+```
+
+### To update the remote master branch:
+
+> you have to be able to push to `master` branch to complete this operation
```
Comment (tomekpapiernik) [inline CONTRIBUTING.md]:
```suggestion
>**NOTE:** You must have permissions to push to the `master` branch to complete this operation.
```
Which master branch? Kyma? Dex?

File: CONTRIBUTING.md (hunk 13 above)
Comment (tomekpapiernik) [inline CONTRIBUTING.md]: Additional comment - I think it's better to have steps as numbered lists instead of commented lines in code blocks.

**Comment by Tomasz-Smelcerz-SAP** [comment]
This is stale, we plan to remove dex soon.

**Pull request closed without merging**